inference/core/workflows/core_steps/sinks/roboflow/roboflow_custom_metadata.py

```diff
@@ -0,0 +1,220 @@
+
+For more information on Model Monitoring at Roboflow, see https://docs.roboflow.com/deploy/model-monitoring.
+"""
+
+WORKSPACE_NAME_CACHE_EXPIRE = 900  # 15 min
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "sink",
+        }
+    )
+    type: Literal["RoboflowCustomMetadata"]
+    predictions: StepOutputSelector(
+        kind=[
+            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+            BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
+            BATCH_OF_CLASSIFICATION_PREDICTION_KIND,
+        ]
+    ) = Field(
+        description="Reference data to extract property from",
+        examples=["$steps.my_step.predictions"],
+    )
+    field_value: StepOutputSelector(kind=[STRING_KIND]) = Field(
+        description="This is the name of the metadata field you are creating",
+        examples=["toronto", "pass", "fail"],
+    )
+    field_name: str = Field(
+        description="Name of the field to be updated in Roboflow Customer Metadata",
+        examples=["The name of the value of the field"],
+    )
+    fire_and_forget: Union[bool, WorkflowParameterSelector(kind=[BOOLEAN_KIND])] = (
+        Field(
+            default=True,
+            description="Boolean flag dictating if sink is supposed to be executed in the background, "
+            "not waiting on status of registration before end of workflow run. Use `True` if best-effort "
+            "registration is needed, use `False` while debugging and if error handling is needed",
+        )
+    )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="error_status", kind=[BOOLEAN_KIND]),
+            OutputDefinition(name="message", kind=[STRING_KIND]),
+            OutputDefinition(
+                name="predictions", kind=[BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND]
+            ),
+        ]
+
+
+class RoboflowCustomMetadataBlock(WorkflowBlock):
+
+    def __init__(
+        self,
+        cache: BaseCache,
+        api_key: Optional[str],
+        background_tasks: Optional[BackgroundTasks],
+    ):
+        self._api_key = api_key
+        self._cache = cache
+        self._background_tasks = background_tasks
+
+    @classmethod
+    def get_init_parameters(cls) -> List[str]:
+        return ["api_key", "cache", "background_tasks"]
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    async def run(
+        self,
+        fire_and_forget: bool,
+        field_name: str,
+        field_value: StepOutputSelector(kind=[BATCH_OF_STRING_KIND]),
```

Comment:
I think the type here should be `Batch[str]`
Similarly, below the type should probably be `Batch[sv.Detections]`